### PR TITLE
[Rule Tuning] Tuning `Account Password Reset Remotely`

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/detection-rules/issues/3473

## Summary
This pull request attempts to tune the `Account Password Reset Remotely` rule for performance related to the issue above. The following has been considered when tuning:

* Does the EQL sequence have a maxspan; does the maxspan have an adequate time defined
* If this is an EQL sequence, can it be reduced to a non-sequence EQL query
* If this is an EQL sequence, is the first sequence refined enough memory consumption
* Replace any wildcards with EQL functions
* If unable to replace wildcard usage, leverage built-in regex for pattern matching
* Review logic and determine if it can be adjusted to tighten the scope of the intended threat attempting to be detected

The `note` value has also been updated to be more informative about the expectation of this rule being noisy and the recommendation to duplicate and adjust accordingly.

### CODE REVIEW

**Comment by terrancedejesus** [comment]
Please note - This tuning pull request will remain in draft until we test and confirm performance increases with the suggested rule change. We also need to confirm there is no regression introduced for the logic of the query.

File: rules/windows/persistence_remote_password_reset.toml [hunk 1]
```diff
@@ -34,19 +36,30 @@ type = "eql"
...
+
     /* event 4624 need to be logged */
     winlog.logon.type : "Network" and event.outcome == "success" and source.ip != null and
     source.ip != "127.0.0.1" and source.ip != "::1" and
-    not winlog.event_data.TargetUserName : ("svc*", "PIM_*", "_*_", "*-*-*", "*$")] by winlog.event_data.TargetLogonId
+    not startsWith~(winlog.event_data.TargetUserName, "svc") and
+    not startsWith~(winlog.event_data.TargetUserName, "PIM_") and
+    not endsWith~(winlog.event_data.TargetUserName, "$") and
+    not (startsWith(winlog.event_data.TargetUserName, "_") and endsWith(winlog.event_data.TargetUserName, "_")) and
+    not winlog.event_data.TargetUserName regex "[a-zA-Z0-9]+-[a-zA-Z0-9]+-[a-zA-Z0-9]+"] by winlog.event_data.TargetLogonId
```
Comment (w0rk3r) [inline rules/windows/persistence_remote_password_reset.toml:47]: What are we excluding here?

File: rules/windows/persistence_remote_password_reset.toml [hunk 2]
```diff
@@ -34,19 +36,30 @@ type = "eql"
...
-    winlog.event_data.TargetUserName: ("*Admin*", "*super*", "*SVC*", "*DC0*", "*service*", "*DMZ*", "*ADM*") or
-    winlog.event_data.TargetSid : ("S-1-5-21-*-500", "S-1-12-1-*-500")
+    stringContains(winlog.event_data.TargetUserName, "Admin") or
+    stringContains(winlog.event_data.TargetUserName, "super") or
+    stringContains(winlog.event_data.TargetUserName, "SVC") or
+    stringContains(winlog.event_data.TargetUserName, "DC0") or
+    stringContains(winlog.event_data.TargetUserName, "service") or
+    stringContains(winlog.event_data.TargetUserName, "DMZ") or
+    stringContains(winlog.event_data.TargetUserName, "ADM") or
+    winlog.event_data.TargetSid regex~ "S-1-(5-21|12-1)-.*-500"
```
Comment (Samirbous) [inline rules/windows/persistence_remote_password_reset.toml:62]: TargetSid is always uppercase + numbers
```suggestion
    winlog.event_data.TargetSid regex "S-1-(5-21|12-1)-.*-500"
```

File: rules/windows/persistence_remote_password_reset.toml (hunk 1 above)
Comment (Samirbous) [reply rules/windows/persistence_remote_password_reset.toml:47]: the above are most noisy FPs based on service account naming convention

File: rules/windows/persistence_remote_password_reset.toml [hunk 3]
```diff
@@ -18,7 +18,9 @@ index = ["winlogbeat-*", "logs-system.security*", "logs-windows.forwarded*"]
 language = "eql"
 license = "Elastic License v2"
 name = "Account Password Reset Remotely"
-note = "This rule may cause medium to high performance impact due to logic scoping all remote Windows logon activity."
+note = """
+This rule may cause medium to high performance impact due to scoping all remote Windows logon activity within the first sequence. This rule by default will be noisy if not scoped to custom privileged accounts. If noisy, duplicate the rule and replace the `TargetUserName` or `TargetSid` filters with your own naming convention and accounts of interest.
```
Comment (Samirbous) [inline rules/windows/persistence_remote_password_reset.toml:22]: maybe add the `## Performance` header ? also adding a performance to the tags array will help filter rules via GUI.

Comment (terrancedejesus) [reply rules/windows/persistence_remote_password_reset.toml:22]: Added performance header. I'd like to discuss with the team more about tagging options. Do not want to block release.

**Comment by terrancedejesus** [comment]
> Please note - This tuning pull request will remain in draft until we test and confirm performance increases with the suggested rule change. We also need to confirm there is no regression introduced for the logic of the query.

This note was added to all potential performance tunings PR from a recent SDH. We were given access to this stack and tested this query by duplicating the OOTB rule and adjusting the query. The rule ran successfully with no issues, however RCA did not determine that performance was related to the query itself.

**Comment by terrancedejesus** [comment]
For the rule versions to be fixed for this rule, it will require a min-stack of 8.13 to fork the rule and versions. Min-stack has been updated accordingly.

**Comment by terrancedejesus** [comment]
Proposed query for historical records aside from commit.

```
sequence by winlog.computer_name with maxspan=1m
  [authentication where event.action == "logged-in" and

    /* event 4624 need to be logged */
    winlog.logon.type : "Network" and event.outcome == "success" and source.ip != null and
    source.ip != "127.0.0.1" and source.ip != "::1" and
    not startsWith~(winlog.event_data.TargetUserName, "svc") and
    not startsWith~(winlog.event_data.TargetUserName, "PIM_") and
    not endsWith~(winlog.event_data.TargetUserName, "$") and
    not (startsWith(winlog.event_data.TargetUserName, "_") and endsWith(winlog.event_data.TargetUserName, "_")) and
    not winlog.event_data.TargetUserName regex "[a-zA-Z0-9]+-[a-zA-Z0-9]+-[a-zA-Z0-9]+"] by winlog.event_data.TargetLogonId
   /* event 4724 need to be logged */
  [iam where event.action == "reset-password" and
   (
    /*
       This rule is very noisy if not scoped to custom privileged accounts.
       If noisy, duplicate the rule and replace the filters below with your own naming convention and accounts of interest.
     */
    stringContains(winlog.event_data.TargetUserName, "Admin") or
    stringContains(winlog.event_data.TargetUserName, "super") or
    stringContains(winlog.event_data.TargetUserName, "SVC") or
    stringContains(winlog.event_data.TargetUserName, "DC0") or
    stringContains(winlog.event_data.TargetUserName, "service") or
    stringContains(winlog.event_data.TargetUserName, "DMZ") or
    stringContains(winlog.event_data.TargetUserName, "ADM") or
    winlog.event_data.TargetSid regex "S-1-(5-21|12-1)-.*-500"
    )
  ] by winlog.event_data.SubjectLogonId
```

**Pull request merged**